src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -90,7 +90,8 @@ bool IsSuppressedOnElement (int id, ICustomAttributeProvider provider, out Suppr
 				return false;
 
 			return _suppressions.TryGetValue (provider, out var suppressions) &&
-				suppressions.TryGetValue (id, out info);
+				suppressions.TryGetValue (id, out info) &&
+				(provider is ModuleDefinition || provider is AssemblyDefinition ? info.Target == null : true);
```

Comment:
Why is this necessary - the code in `LookForModuleLevelAndGlobalSuppressions` will only add the suppression to the list if
* Target is null and provider is module or assembly
* Target is not null, but then provider is not module or assembly

In which case do we need this additional condition?

Reply:
Good call. I was hitting this whenever a global suppression was being added from `MarkStep` https://github.com/mono/linker/blob/41fd293dda27c099b063f51d8917e607271a508e/src/linker/Linker.Steps/MarkStep.cs#L811-L812
Changed things there so that that path isn't followed for global suppressions, otherwise we'd hit this (having suppressions where the provider is a module/assembly and it has a target).